### PR TITLE
New CLI top-level section

### PR DESCRIPTION
**Pull request by Tonnu**
This PR moves the "using the cli" section from the docs to a new "CLI" section, found in the top navbar. It also introduces some new articles and new article structure.

### CODE REVIEW

**Comment by flenter** [comment]
Apart from the failing build, I think this causes some issues for the menu when viewing the site on mobile. Or can't people get to the cli section on mobile?

The menu I'm thinking of currently looks like this on devcenter.wercker.com:
screen shot 2016-01-12 at 12 42 01

File: docs-toc-menu/index.js [hunk 1]
```diff
@@ -31,6 +31,9 @@ module.exports = react.createClass({
       case 'quickstarts':
         data = 'Quickstarts';
         break;
+			case 'cli':
```
Comment (flenter) [inline docs-toc-menu/index.js]: The indenting is a bit quirky here

File: docs-sidebar/toc-cli.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+[
+  [
+    "installation"
+  ], 
+  [
+    "usage",
+    "developing.md",
+    "building.md",
+		"pulling-builds.md",
```
Comment (flenter) [inline docs-sidebar/toc-cli.json]: indentation is a bit off here

File: docs-sidebar/index.js [hunk 3]
```diff
@@ -26,6 +27,9 @@ function createClass () {
         case 'quickstarts':
           data = quickstartsToc;
           break;
+				case 'cli':
```
Comment (flenter) [inline docs-sidebar/index.js]: Indentation

File: docs-nav/index.js [hunk 4]
```diff
@@ -32,6 +32,9 @@ module.exports = react.createClass({
       case 'quickstarts':
         data = 'quickstarts';
         break;
+			case 'cli':
```
Comment (flenter) [inline docs-nav/index.js]: indentation?

File: docs-header/index.js [hunk 5]
```diff
@@ -26,6 +26,9 @@ module.exports = react.createClass({
       case 'quickstarts':
         data = 'Quickstarts';
         break;
+			case 'cli':
```
Comment (flenter) [inline docs-header/index.js]: indentation is a bit off here

**Comment by flenter** [comment]
Is there a way that we can add redirects, so people are less likely to get 404's when arriving from a search engine?

**Comment by hatchan** [comment]
There's the legacy-urls module: https://github.com/wercker/docs/tree/master/legacy-urls That will add redirect the first url to the second url.

File: content/cli/configuration/options-and-flags.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+the container to the host.
+
+`wercker dev --publish 1337`
+
+#### <a name="no-remove" class="anchor"></a>--no-remove
+
+When developing locally, you can opt not to remove the development containers
+locally.
+
+#### <a name="store-s3" class="anchor"></a>--store-s3
```
Comment (hatchan) [inline content/cli/configuration/options-and-flags.md]: Not sure if we should document this, as this is just used internally for our hosted service.

File: content/cli/configuration/options-and-flags.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+
+You can store your container images and artefacts on Amazon S3 while building
```
Comment (hatchan) [inline content/cli/configuration/options-and-flags.md]: If we are going to keep this, we should probably rename `artefacts` to `artifacts`

File: content/cli/usage/building.md [hunk 8]
```diff
@@ -24,5 +24,15 @@ to save the container you must commit it first:
...
 ```
 
-The container will be committed to the the Docker host after which you
-can run it.
+### internal/docker-push
+
+The container will be committed to the the Docker host after which you can run
+it. Alternatively, if using the
+[internal/docker-push](/docs/steps/internal-steps.html#docker-push) together
+with the [docker-local](/cli/configuration/options-and-flags.html#docker-local) flag, the build container
```
Comment (hatchan) [inline content/cli/usage/building.md]: I think this should say, that it is __only__ going to be committed to the local Docker daemon.

**Comment by lindseybateman** [comment]
See this PR on staging 
http://devcenter-staging.wercker.com/cli/index.html

**Comment by Tonnu** [comment]
closing this in favor of #342 due to rebasing issuess :(

**Pull request closed without merging**